src/Components/Server/src/ComponentHub.cs

```diff
@@ -45,23 +61,22 @@ internal sealed class ComponentHub : Hub
         /// </summary>
         public static PathString DefaultPath { get; } = "/_blazor";
 
-        /// <summary>
-        /// For unit testing only.
-        /// </summary>
-        // We store the circuit host in Context.Items which is tied to the lifetime of the underlying
-        // SignalR connection. There's no need to clean this up, it's a non-owning reference and it
-        // will go away when the connection does.
-        internal CircuitHost CircuitHost
+        // We store the CircuitHost through a *handle* here because Context.Items is tied to the lifetime
+        // of the connection. It's possible that a misbehaving client could cause disposal of a CircuitHost
+        // but keep a connection open indefinitely, preventing GC of the Circuit and related application state.
+        // Using a handle allows the CircuitHost to clear this reference in the background.
```

Comment:
Where does it get cleared? I'm struggling to see anything that writes to either `.CircuitHost` or to the property inside the `CircuitHandle`. What obvious thing am I missing?

Reply:
It's here? https://github.com/aspnet/AspNetCore/pull/12857/files#diff-8c77562947e4970a8d836d0df8be2b79R169

Reply:
Yes, thanks. I was missing this repeatedly because of how GitHub collapses large diffs, so my attempts to find code that writes to this property were returning no results. This PR was much easier to read once I checked out the sources locally.